frontend/src/features/map_planning/routes/Map.tsx

```diff
@@ -11,39 +12,38 @@ import { Circle, Rect } from 'react-konva';
  * Otherwise they cannot be moved.
  */
 export const Map = () => {
-  const [shapes, setShapes] = useState<JSX.Element[]>([]);
+  const state = useMapStore((map) => map.state);
 
-  useEffect(() => {
-    const newShapes = [];
-    for (let i = 1; i <= 10; i++) {
-      newShapes.push(
-        <Circle
-          key={i}
-          x={100 * i}
-          y={100 * i}
-          radius={50}
-          fill="red"
-          draggable={true}
-          shadowBlur={5}
-        />,
-        <Rect
-          key={i + 100}
-          x={100 * i + 200}
-          y={100 * i + 10}
-          width={100}
-          height={100}
-          fill="blue"
-          draggable={true}
-          shadowBlur={5}
-        />,
-      );
+  // Event listener responsible for adding a single shape to the transformer
+  const addToTransformer = (node: Shape<ShapeConfig>) => {
+    const transformer = useMapStore.getState().transformer.current;
+
+    const nodes = transformer?.getNodes() || [];
+    if (!nodes.includes(node)) {
+      transformer?.nodes([node]);
     }
-    setShapes(newShapes);
-  }, []);
+  };
 
   return (
     <BaseStage>
-      <PlantsLayer>{shapes}</PlantsLayer>
+      <PlantsLayer>
+        {state.layers['plant'].objects.map((o) => (
+          <Rect
+            {...o}
+            key={o.id}
+            fill="blue"
+            draggable={true}
+            shadowBlur={5}
+            onClick={(e) => {
+              addToTransformer(e.target as Shape<ShapeConfig>);
+            }}
+            onDragStart={(e) => {
+              // sometimes the click event is not fired, so we have to add the object to the transformer here
```

Comment:
Is the object added twice when both events fire?

Reply:
No, the reference would be equal and only the node from the first event is added to the transformer. The second can not pass the check in line 22
```ts
if (!nodes.includes(node)) {
  transformer?.nodes([node]);
}
```

Reply:
makes sense